vars/addRcDetailsComment.groovy

```diff
@@ -17,22 +17,22 @@ import jenkins.ReleaseMetricsData
  * */
 void call(Map args = [:]) {
     def buildIndexName = 'opensearch-distribution-build-results'
-    def version = args.version
-    def qualifier = "None"
-    def matcher = version =~ /^([\d.]+)(?:-(.+))?$/
-    if (matcher) {
-        version = matcher[0][1]  // Captures the numeric part (3.0.0)
-        qualifier = matcher[0][2] ?: "None" // Captures the qualifier (beta1) or None if no qualifier
+    if (args.version.isEmpty()){
+        error('version is required to get RC details.')
     }
+    def (version, qualifier) = { localVersion ->
+        def localMatcher = localVersion =~ /^([\d.]+)(?:-(.+))?$/
```

Comment:
Is it easier to just use tokenizer?
https://github.com/opensearch-project/ml-commons/blob/main/build.gradle#L19

Reply:
Let me try that.

Reply:
Updated the code. Thanks!